httpclient/httpclient.go

```diff
@@ -175,23 +169,18 @@ func (c *Client) Call(ctx context.Context, r Request) (err error) {
 		return req, nil
 	}
 
-	return c.retryRequest(ctx, r, newRequestFn)
+	return c.retryRequest(ctx, spanName, r, newRequestFn)
 }
 
 // retryRequest will make the request and only call the decoder when a 2XX has been received.
 // Any response body in non 2XX cases is discarded.
 // nolint: funlen
-func (c *Client) retryRequest(ctx context.Context, r Request, newRequestFn func() (*http.Request, error)) error {
-	finalStatus := 0
-	defer func() {
-		// Add the status code from after all retries to the parent span
-		o11y.FromContext(ctx).GetSpan(ctx).AddRawField("http.status_code", finalStatus)
-	}()
-
+func (c *Client) retryRequest(
+	ctx context.Context, spanName string, r Request, newRequestFn func() (*http.Request, error)) error {
```

Comment:
~ can you break the line after the context

https://github.com/circleci/engineering/blob/master/howto/golang.md#wrapping-long-lines